src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs

```diff
@@ -345,13 +343,11 @@ private async ValueTask<bool> TryGetProjectDiagnosticsFromInMemoryStorageAsync(V
                 return false;
             }
 
-            private ValueTask<ImmutableArray<DiagnosticData>> GetDiagnosticsFromInMemoryStorageAsync(
```

Comment:
>GetDiagnosticsFromInMemoryStorageAsync [](http://example.com/codeflow?start=62&length=38)

seems like something should flag private Task returning methods that execute synchronously.

Reply:
yes. t hough it can be a bit of a challenge.  imagine:

```c#
ValueTask<bool> DoSomethingAsync()
{
    if (x)
        return new(0);

    return SomethingElseAsync();
}
```

All synchronous, all entirely fine.